components/provisioner/internal/hydroform/configuration.go

```diff
@@ -47,36 +42,67 @@ func buildConfigForGCP(config model.GCPConfig, credentialsFile string) (*types.C
 }
 
 func buildConfigForGardener(config model.GardenerConfig, credentialsFile string) (*types.Cluster, *types.Provider, error) {
-	diskSize, err := strconv.Atoi(config.VolumeSize)
-
-	if err != nil {
-		return &types.Cluster{}, &types.Provider{}, err
-	}
 
 	cluster := &types.Cluster{
 		KubernetesVersion: config.KubernetesVersion,
 		Name:              config.Name,
-		DiskSizeGB:        diskSize,
+		DiskSizeGB:        config.VolumeSizeGB,
 		NodeCount:         config.NodeCount,
 		Location:          config.Region,
 		MachineType:       config.MachineType,
 	}
 
+	customConfiguration := map[string]interface{}{
+		"target_provider": config.Provider,
+		"target_seed":     config.Seed,
+		"target_secret":   config.TargetSecret,
+		"disk_type":       config.DiskType,
+		"workercidr":      config.WorkerCidr,
+		"autoscaler_min":  config.AutoScalerMin,
+		"autoscaler_max":  config.AutoScalerMax,
+		"max_surge":       config.MaxSurge,
+		"max_unavailable": config.MaxUnavailable,
+	}
+
+	customConfiguration, err := addProviderSpecificConfig(customConfiguration, config.ProviderSpecificConfig)
+
+	if err != nil {
+		return &types.Cluster{}, &types.Provider{}, err
+	}
+
 	provider := &types.Provider{
-		Type:                types.Gardener,
-		ProjectName:         config.ProjectName,
-		CredentialsFilePath: credentialsFile,
-		CustomConfigurations: map[string]interface{}{
-			"target_provider": config.TargetProvider,
-			"target_secret":   config.TargetSecret,
-			"disk_type":       config.DiskType,
-			"zone":            config.Zone,
-			"cidr":            config.Cidr,
-			"autoscaler_min":  config.AutoScalerMin,
-			"autoscaler_max":  config.AutoScalerMax,
-			"max_surge":       config.MaxSurge,
-			"max_unavailable": config.MaxUnavailable,
-		},
+		Type:                 types.Gardener,
+		ProjectName:          config.ProjectName,
+		CredentialsFilePath:  credentialsFile,
+		CustomConfigurations: customConfiguration,
 	}
 	return cluster, provider, nil
 }
+
+func addProviderSpecificConfig(customConfiguration map[string]interface{}, providerSpecificConfig string) (map[string]interface{}, error) {
```

Comment:
If we look at operation flow, the following happens:
- graphQL structures are converted to model structures (provider specific config is encoded to json)
- model structures are converted to hydroform specific format (provider specific config is decoded from json)

It looks a bit strange at the first glance: first we encode to json and then decode. 
So the following questions arise: 
- should hydroform package depend on model? 
- should types defined in  model be moved to persistence package? 
  
What we could do is:
-  moving types defined in model package to persistence package. The reasoning for that is strong relation between types defined in the model package and database design.
- changing hydroform to use gqlschema package. The reasoning for that is that we will avoid additional json decoding and simplify model (GCPProviderConfig, AWSProviderConfig and AzureProviderConfig will no longer be needed)